integration/shareddata/benchmarks.go

```diff
@@ -15,61 +15,68 @@
 package shareddata
 
 import (
-	"math/rand"
-	"time"
-
 	"go.mongodb.org/mongo-driver/bson"
-	"go.mongodb.org/mongo-driver/bson/primitive"
 	"golang.org/x/exp/maps"
 )
 
-// BenchmarkSmallDocuments provides 10000 documents that look like:
+// BenchmarkSmallDocuments provides documents that look like:
 //
 //	{_id: int32(0), id: int32(0), v: "foo"}
 //	{_id: int32(1), id: int32(1), v: int32(42)}
 //	{_id: int32(2), id: int32(2), v: "42"}
 //	{_id: int32(3), id: int32(3), v: {"foo": int32(42)}}
 //	...
 //
-// `_id` is a primary key that goes from 0 to n-1.
+// `_id` is an int32 primary key that starts from 0.
 // `id` has the same value as `_id`, but is not indexed by default.
 // `v` has one of the four values shown above.
-var BenchmarkSmallDocuments = newGeneratorBenchmarkProvider("SmallDocuments", 10000, smallDocumentsGenerator)
-
-// BenchmarkLessSmallDocuments provides 100 documents that look like BenchmarkSmallDocuments.
-var BenchmarkLessSmallDocuments = newGeneratorBenchmarkProvider("LessSmallDocuments", 100, smallDocumentsGenerator)
-
-// BenchmarkLargeDocuments provides a single large document with 123 fields of various types that consists
-// of different long keys and simple values.
-var BenchmarkLargeDocuments = newGeneratorBenchmarkProvider("LargeDocuments", 123, func(n int) generatorFunc {
+var BenchmarkSmallDocuments = newGeneratorBenchmarkProvider("SmallDocuments", func(docs int) generatorFunc {
 	values := []any{
-		true, "foo", false, true, 512, true, 42, false, -42, "42", false, false, true,
-		bson.D{{"42", "hello"}},
-		primitive.NewDateTimeFromTime(time.Date(2021, 11, 1, 10, 18, 42, 123000000, time.UTC)),
-		false, true, 42.13, 501,
+		"foo", int32(42), "42", bson.D{{"foo", int32(42)}},
 	}
-	vLen := len(values)
+	l := len(values)
 
-	elements := make([]bson.E, n)
-	elements[0] = bson.E{"_id", 0}
+	var total int
 
-	for i := 1; i < len(elements); i++ {
-		elements[i] = bson.E{
-			Key:   genLetters(i, 50),
-			Value: values[i%vLen],
+	return func() bson.D {
```

Comment:
I see, each time caller of this function calls it, it returns `values` from next index.